go/state/state_db.go

```diff
@@ -580,6 +595,24 @@ func (s *stateDB) Suicide(addr common.Address) bool {
 	return true
 }
 
+// SuicideNewContract marks the given contract account as suicided.
+// If called in the same transaction scope after CreateContract, works
+// same as Suicide, otherwise has no effect.
+func (s *stateDB) SuicideNewContract(addr common.Address) bool {
+	if !s.Exist(addr) {
```

Comment:
Since the "Exist" is also tested in the `Suicide` you might skip it here and just do the following:
```
if s.createdContracts[addr] {
    return s.Suicide(addr)
}
return false
```